src/gpu/sycl/ref_prelu.cpp

```diff
@@ -0,0 +1,147 @@
+* limitations under the License.
+*******************************************************************************/
+
+#include "gpu/sycl/ref_prelu.hpp"
+#include "common/utils.hpp"
+#include "gpu/sycl/prelu_kernels.hpp"
+
+#include <typeinfo>
+#include "gpu/nvidia/sycl_cuda_utils.hpp"
+namespace dnnl {
+namespace impl {
+namespace gpu {
+namespace sycl {
+
+using namespace impl::sycl;
+
+status_t ref_prelu_fwd_t::pd_t::init_conf() {
+    if (has_zero_dim_memory()) return status::success;
+
+    conf_ = sycl_prelu_conf_t();
+
+    const memory_desc_wrapper data_d(src_md(0));
+    const memory_desc_wrapper weights_d(weights_md(0));
+    conf_.data_md = sycl_md_t(src_md(0));
+    conf_.weights_md = sycl_md_t(weights_md(0));
+    conf_.dst_md = sycl_md_t(dst_md(0));
+    conf_.ndims = ndims();
+    conf_.mask = utils::get_dims_mask(data_d.dims(), weights_d.dims(), ndims());
+
+    conf_.block_size = 16;
+    conf_.wg_size = 32;
+    conf_.work_amount = memory_desc_wrapper(src_md(0)).nelems();
+    conf_.work_amount_wei = memory_desc_wrapper(weights_md(0)).nelems();
+    int work_per_wg = conf_.wg_size * conf_.block_size;
+    int n_wgs = (conf_.work_amount + work_per_wg - 1) / work_per_wg;
+    conf_.n_thr = n_wgs * conf_.wg_size;
+
+    return status::success;
+}
+
+status_t ref_prelu_fwd_t::init(engine_t *engine) {
+    const auto kid = ::sycl::get_kernel_id<prelu_fwd_kernel_vec_t>();
+    CHECK(create_kernel(engine, kid, &kernel_));
+
+    return status::success;
+}
+
+status_t ref_prelu_fwd_t::execute_forward(const exec_ctx_t &ctx) const {
+    if (pd()->has_zero_dim_memory()) return status::success;
+
+    parallel_for(ctx, kernel_, [&](::sycl::handler &cgh) {
+        auto data = CTX_IN_SYCL_KERNEL_MEMORY(DNNL_ARG_SRC);
+        auto weights = CTX_IN_SYCL_KERNEL_MEMORY(DNNL_ARG_WEIGHTS);
+        auto dst = CTX_OUT_SYCL_KERNEL_MEMORY(DNNL_ARG_DST);
+        auto nelems_A = memory_desc_wrapper(pd()->src_md(0)).nelems();
+        int tot_work = nelems_A;
+        prelu_fwd_kernel_vec_t prelu_fwd_kernel(
+                pd()->conf_, data, weights, dst);
+        const int block_size = pd()->conf_.block_size;
+        const int wg_size = pd()->conf_.wg_size;
+        int work_per_wg = wg_size * block_size;
+        int n_wgs = (tot_work + work_per_wg - 1) / work_per_wg;
+        int n_thr = n_wgs * wg_size;
+        cgh.parallel_for(::sycl::nd_range<1>(n_thr, wg_size), prelu_fwd_kernel);
+    });
+
+    return status::success;
+}
+
+//prelu backward
+
+status_t ref_prelu_bwd_t::pd_t::init_conf() {
+    if (has_zero_dim_memory()) return status::success;
+    conf_ = sycl_prelu_conf_t();
+    conf_.data_md = sycl_md_t(src_md(0));
+    conf_.weights_md = sycl_md_t(weights_md(0));
+    conf_.diff_data_md = sycl_md_t(diff_src_md(0));
+    conf_.diff_weights_md = sycl_md_t(diff_weights_md(0));
+    conf_.diff_dst_md = sycl_md_t(diff_dst_md(0));
+    conf_.dst_md = sycl_md_t(dst_md(0));
```

Comment:
dst_md is not defined for backward pass, this causes an assert hit in debug mode

Reply:
Removed.